hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestMergeIntoTable2.scala

```diff
@@ -884,10 +887,9 @@ class TestMergeIntoTable2 extends HoodieSparkSqlTestBase {
          """.stripMargin
           )
           checkAnswer(s"select id, name, price, ts, dt from $tableName")(
-            Seq(1, "a1", 10.1, 1000, "2021-03-21"),
             Seq(1, "a2", 10.2, 1002, "2021-03-21"),
-            Seq(3, "a3", 10.3, 1003, "2021-03-21"),
-            Seq(1, "a2", 10.2, 1002, "2021-03-21")
```

Comment:
Slight behavior change here. Previously we were doing an insert when matched because of no precombine key. Now we actually do an update

Reply:
after 2nd MIT, for id 1, should we see just 2 records?
after 1st MIT command, storage has two records for id 1. and in the 2nd MIT command, id 1 matches and so we might have to do update. So, the same two records on storage should get updated right? why do we see 3 records for id 1 ? can you throw some light please.

Reply:
The original behavior was to do an insert in that situation which is why there were 3 records for id 1. After my changes there were actually 4 records with id 1 because we matched both the id 1 records on the join.